libmproxy/web/app.py

```diff
@@ -240,10 +277,14 @@ def put(self):
 
 
 class Application(tornado.web.Application):
-    def __init__(self, master, debug):
+    def __init__(self, master, debug, singleuser, htpasswd):
         self.master = master
+        self.additional_args = dict(
+            singleuser=singleuser,
+            htpasswd=htpasswd,
+        )
         handlers = [
-            (r"/", IndexHandler),
+            (r"/", IndexHandler, self.additional_args),
```

Comment:
So, this looks like it only implements basic auth for the index page, but not the API?